package_scanner.nim

```diff
@@ -1,135 +1,367 @@
-    "BSD",
-    "BSD2",
-    "BSD3",
-    "CC0",
-    "GPL",
-    "GPLv2",
-    "GPLv3",
-    "LGPLv2",
-    "LGPLv3",
-    "MIT",
-    "MS-PL",
-    "MPL",
-    "WTFPL",
+  allowBrokenUrl = true      ## Allow broken Repo URLs
+  allowMissingNimble = true  ## Allow missing ``*.nimble`` files
+  checkNimbleFile = true     ## Check if repos have ``*.nimble`` file
+  checkProfanity = false     ## Check for Profanities
+  checkByGit = false         ## Check via Git
+  httpTimeout = 10_000       ## Timeout. Below ~2000 false positives happen?
+  tagsMaxLen = 32            ## Maximum lenght for the string in tags
+  tagsMaximum = 17           ## Maximum number of tags allowed on the list
+  lineLenMax = 250           ## Maximum line lenght for values
+  profanityLevensteinMax = 1 ## Levenstein Distance algo threshold
+  vcsTypes = ["git", "hg"]   ## Valid known Version Control Systems.
+  tagsBlacklist = ["nimrod", "nim"] ## Tags that should be not allowed.
+  keysRequired = ["name", "url", "method", "tags", "description", "license"]
+  gitTempDir = "/tmp/gitTempDir"
+  gitCmd = "git clone --no-checkout --no-tags --bare --depth 1 $1 " & gitTempDir
+  nodeCmd = "node ./validate_json.js"
+  pythonCmd = "python3 ./validate_json.py"
+  rubyCmd = "ruby ./validate_json.rb"
+  defaultFilePermissions = {fpUserWrite, fpUserRead, fpGroupWrite, fpGroupRead, fpOthersRead}
+  packagesFilePath = currentSourcePath().parentDir / "packages.json"
+  packagesJsonStr = readFile(packagesFilePath)
+  hostsSkip = [
+    "https://bitbucket",
+    "https://gitlab.3dicc",
+    "https://mahlon@bitbucket",
+    "https://notabug",
+  ] ## Hostnames to skip checks, for reliability, wont like direct HTTP GET?.
+  licenses = [
+    "allegro 4 giftware",
+    "apache",
+    "apache2",
+    "apache2.0",
+    "apache 2.0",
+    "apache-2.0",
+    "apache version 2.0",
+    "apache license 2.0",
+    "bsd",
+    "bsd2",
+    "bsd-2",
+    "bsd3",
+    "bsd-3",
+    "bsd 2-clause",
+    "bsd-2-clause",
+    "bsd 3-clause",
+    "bsd-3-clause",
+    "cc0",
+    "gpl",
+    "gpl2",
+    "gpl-2.0",
+    "gpl3",
+    "gpl-3.0",
+    "gplv2",
+    "gplv3",
+    "lgpl",
+    "lgplv2",
+    "lgplv2.1",
+    "lgplv3",
+    "mit",
+    "ms-pl",
+    "mpl",
+    "ppl",
+    "wtfpl",
     "libpng",
     "zlib",
-    "ISC",
-    "Unlicense"
-  ]
-  VCS_TYPES = @["git", "hg"]
+    "isc",
+    "unlicense",
+    # Not concrete Licenses, but some rare cases observed on the JSON.
+    "0bsd",
+    "agplv3",
+    "agpl-3.0",
+    "apache license 2.0 or mit",
+    "fontconfig",
+    "gnu lesser general public license v2.1",
+    "public domain",
+    "lgplv3 or gplv2",
+    "openssl and ssleay",
+    "openldap",
+    "apache 2.0 or gplv2",
+    "mit or apache 2.0",
+    "lgpl with static linking exception",
+  ]  ## All valid known licences for Nimble packages, on lowercase.
+when checkProfanity:
+  const profanitiesStr = readFile(currentSourcePath().parentDir / "profanities.txt").normalize.splitLines
```

Comment:
This is cool, but I don't think it's necessary, in fact if people want to call their package "fuck" I have no problem with that. I think humans can reject anything more offensive.

Reply:
is compile time disabled, if some day want to enable it. :slightly_smiling_face:

Reply:
It's more code to be maintained...